src/main/java/io/cdap/plugin/http/source/common/BaseHttpSourceConfig.java

```diff
@@ -0,0 +1,931 @@
+  @Macro
+  protected String httpMethod;
+
+  @Name(PROPERTY_HEADERS)
+  @Nullable
+  @Description("Headers to send with each HTTP request.")
+  @Macro
+  protected String headers;
+
+  @Nullable
+  @Name(PROPERTY_REQUEST_BODY)
+  @Description("Body to send with each HTTP request.")
+  @Macro
+  protected String requestBody;
+
+  @Name(PROPERTY_FORMAT)
+  @Description("Format of the HTTP response. This determines how the response is converted into output records.")
+  @Macro
+  protected String format;
+
+  @Nullable
+  @Name(PROPERTY_RESULT_PATH)
+  @Description("Path to the results. When the format is XML, this is an XPath. " +
+    "When the format is JSON, this is a JSON path.")
+  @Macro
+  protected String resultPath;
+
+  @Nullable
+  @Name(PROPERTY_FIELDS_MAPPING)
+  @Description("Mapping of fields in a record to fields in retrieved element. The left column contains the " +
+    "name of schema field. The right column contains path to it within a relative to an element. " +
+    "It can be either XPath or JSON path.")
+  @Macro
+  protected String fieldsMapping;
+
+  @Nullable
+  @Name(PROPERTY_CSV_SKIP_FIRST_ROW)
+  @Description("Whether to skip the first row of the HTTP response. " +
+    "This is usually set if the first row is a header row.")
+  @Macro
+  protected String csvSkipFirstRow;
+
+  @Nullable
+  @Name(PROPERTY_USERNAME)
+  @Description("Username for basic authentication.")
+  @Macro
+  protected String username;
+
+  @Nullable
+  @Name(PROPERTY_PASSWORD)
+  @Description("Password for basic authentication.")
+  @Macro
+  protected String password;
+
+  @Nullable
+  @Name(PROPERTY_PROXY_URL)
+  @Description("Proxy URL. Must contain a protocol, address and port.")
+  @Macro
+  protected String proxyUrl;
+
+  @Nullable
+  @Name(PROPERTY_PROXY_USERNAME)
+  @Description("Proxy username.")
+  @Macro
+  protected String proxyUsername;
+
+  @Nullable
+  @Name(PROPERTY_PROXY_PASSWORD)
+  @Description("Proxy password.")
+  @Macro
+  protected String proxyPassword;
+
+  @Nullable
+  @Name(PROPERTY_HTTP_ERROR_HANDLING)
+  @Description("Defines the error handling strategy to use for certain HTTP response codes." +
+    "The left column contains a regular expression for HTTP status code. The right column contains an action which" +
+    "is done in case of match. If HTTP status code matches multiple regular expressions, " +
+    "the first specified in mapping is matched.")
+  protected String httpErrorsHandling;
+
+  @Name(PROPERTY_ERROR_HANDLING)
+  @Description("Error handling strategy to use when the HTTP response cannot be transformed to an output record.")
+  protected String errorHandling;
+
+  @Name(PROPERTY_RETRY_POLICY)
+  @Description("Policy used to calculate delay between retries.")
+  protected String retryPolicy;
+
+  @Nullable
+  @Name(PROPERTY_LINEAR_RETRY_INTERVAL)
+  @Description("Interval between retries. Is only used if retry policy is \"linear\".")
+  @Macro
+  protected String linearRetryInterval;
```

Comment:
You may use not only string types and properties will be converted automatically

Reply:
yes I tried to do that. But that requires me to convert them two times.
1. When I form Mapreduce config I will have to convert those Long to String
2. When I form CDAP config from MR config, I will have to convert String to Long.

If I store them as string I can transform to MR -> CDAP config much simpler.